### DISCUSSION TITLE
Tests: unit tests needed

### DISCUSSION THREAD

**Discussion by coolapso** [streamdeck-linux-gui/streamdeck-linux-gui#51] [Ideas / Feature Requests]
Some tests have been removed as they were relying on a unmaintained  library (hypothesis_auto) created by @timothycrosley which is nowhere to be found (the main reason why this project was forked here). 

In a project like this where many people rely on it releasing stable versions that are functional with less bugs as possible, its important to rely on automated tests to make sure functionalities do no break when changing code, also important that new features are accompanied with the tests for them. 

I know its boring to write tests, and no one really likes them, but they are very much needed. 

If someone has some spare time in their hands and is willing to contribute with some tests, It is very much appreciated! thank you!

**Comment by oparada1988** [comment]
Hi,

I would like to help, im not a programmer, developer myself but certainly have the will power to test! Can you tell me how I can go about setting up the test build? and will it overwrite my current stable build?

**Reply by coolapso** [reply]
Well unit tests are fundamentally different. unit tests test each function of code to make sure it behaves as expected helping to detect breaking changes before they are actually merged into the code. 

But help with actual usage testing is very appreciated. 
you can start reading the documentation here: https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/contributing-guide/#getting-streamdeck_ui-set-up-for-local-development

after the "poetry install" step, you can test the application by doing "poetry run streamdeck"
